### PR TITLE
Fixes #19102 - Add ssh support

### PR DESCRIPTION
**Pull request by hboetes**
This minor edit adds support for ssh urls.

### CODE REVIEW

**Comment by Klaas-** [comment]
works for me (only tested templates import via ssh on tfm-rubygem-foreman_templates-4.0.1-1.fm1_14.el7.noarch)

**Comment by sean797** [comment]
Thanks @hboetes, looks good to me 👍

[test]
ok to test

**Comment by kjelle** [comment]
When we used `foreman_templates` v3.1.0 we used `git@gitlab.host:/group/repo.git`, while in v4.0.1 the only option is `http://`, `https://` and `git://`

From the git-scm docs for `git-clone`:
> The native transport (i.e. git:// URL) does no authentication and should be used with caution on unsecured networks.

In v4.1.0 using deploy keys with git(gub|lab) is no longer possible when syncing templates?

**Comment by kjelle** [comment]
I have manually patched my `/opt/theforeman/tfm/root/usr/share/gems/gems/foreman_templates-4.0.1/app/services/foreman_templates/action.rb` file, adding `ssh://` and it works perfectly!

**Comment by sean797** [comment]
Test failures should have been fixed with theforeman/foreman@807d975. Lets run them again.

[test]

File: app/services/foreman_templates/action.rb [hunk 1]
```diff
@@ -34,7 +34,7 @@ def get_default_branch(repo)
     end
 
     def git_repo?
-      @repo.start_with?('http://', 'https://', 'git://')
+      @repo.start_with?('http://', 'https://', 'git://', 'ssh://')
```
Comment (xprazak2) [inline app/services/foreman_templates/action.rb]: Maybe you could add `ssh+git://` and `git+ssh://` as well.

**Comment by xprazak2** [comment]
@hboetes, I created [issue](http://projects.theforeman.org/issues/19102) in our tracker for this, could you please change commit message to 'Fixes #19102 - Add ssh support', so we can tie these two together?

**Comment by hboetes** [comment]
Sorry, I was rather busy at work with another project.

**Comment by chris1984** [comment]
@hboetes no worries, was just checking.

**Comment by ares** [comment]
[test]

**Comment by ares** [comment]
Thanks @hboetes, I've squash all commits into one and will add other suggested protocols in a separate PR. Merged as 8a0f72be6ecb02e876eed922632e8de624c2e68c

**Pull request closed without merging**